### PR TITLE
[specs] knock some TODOs + review execution engine interaction during derivation

### PR DESCRIPTION
**Pull request by norswap**
build on top of #3208

Changes to the engine interaction part are mostly non-significant (renamed variables, fixed a few sentences).

### CODE REVIEW

File: specs/derivation.md [hunk 1]
```diff
@@ -704,74 +695,109 @@ place. This section explains how this happens.
...
-if (status != "VALID") payload_error()
 
-// retrieve and execute the execution payload
-[executionPayload, rpcErr] = engine_getPayloadV1(payloadID)
-if (rpcErr != null) soft_error()
 
-[status, rpcErr] = engine_newPayloadV1(executionPayload)
-if (rpcErr != null) soft_error()
-if (status != "VALID") payload_error()
+fun makeL2Block(payloadAttributes) {
```
Comment (trianglesphere) [inline specs/derivation.md:716]: is `fun` a typo?

Comment (norswap) [reply specs/derivation.md:716]: Function marker, but could adapt another convention (e.g. `function`) if people feel `fun` is unconventional.

**Review by protolambda — APPROVED** [review]

File: specs/derivation.md [hunk 2]
```diff
@@ -477,40 +484,57 @@ Some frames are ignored:
...
-> - If so, I feel **very strongly** about changing this. This ties us very much to the current implementation.
-> - And it doesn't feel necessary given the channel timeout - if DOS is an issue we can reduce the channel timeout.
-
-> **TODO** The channel queue is a bit weird as implemented (blocks all other channels until the first channel is closed
-> / timed out. Also unclear why we need to wait for channel closure. Maybe something to revisit?
->
-> cf. slack discussion with Proto
+- The channel is pruned out of the channel bank (see below), in which case it isn't passed to the further stages.
+
+> **TODO** specify CHANNEL_TIMEOUT (currently 120s on Goerli testnet)
```
Comment (tynes) [inline specs/derivation.md:511]: Perhaps we put constant values in a table like the consensus specs do?

File: specs/derivation.md [hunk 3]
```diff
@@ -679,74 +694,109 @@ place. This section explains how this happens.
...
-if (status != "VALID") payload_error()
 
-// retrieve and execute the execution payload
-[executionPayload, rpcErr] = engine_getPayloadV1(payloadID)
-if (rpcErr != null) soft_error()
 
-[status, rpcErr] = engine_newPayloadV1(executionPayload)
-if (rpcErr != null) soft_error()
-if (status != "VALID") payload_error()
+fun makeL2Block(payloadAttributes) {
```
Comment (tynes) [inline specs/derivation.md:715]: I prefer using python like syntax throughout the specs but as long as its consistent

Comment (norswap) [reply specs/derivation.md:715]: I'm pretty agnostic about it. Feel free to make a PR to change it, DM me and I'll approve it!

**Review by tynes — APPROVED** [review]

**Pull request merged**